datafusion/sqllogictest/test_files/union.slt

```diff
@@ -135,6 +135,21 @@ SELECT SUM(d) FROM (
 ----
 5
 
+# three way union with aggregate and type coercion
```

Comment:
I verified that this test covers the fix by running this test without the code changes and it fails as expected

```
Running "pg_compat/pg_compat_union.slt"
Running "union.slt"
External error: query failed: DataFusion error: External error: External error: External error: Arrow error: Invalid argument error: RowConverter column schema mismatch, expected Int32 got Int64
[SQL] SELECT c1, SUM(c2) FROM (
    SELECT 1 as c1, 1::int as c2
    UNION
    SELECT 2 as c1, 2::int as c2
    UNION
    SELECT 3 as c1, COALESCE(3::int, 0) as c2
) as a
GROUP BY c1
at test_files/union.slt:139

Error: Execution("1 failures")
error: test failed, to rerun pass `-p datafusion-sqllogictest --test sqllogictests`
```